### PR TITLE
chore: Add linting / format code

### PR DESCRIPTION
**Pull request by dcyriller**
The goal of this PR is to update infrastructure:
- add an eslint / prettier config;
- add a dependabot config;
- drop Bower references;
- drop jshint references.

The only change to the deployment code is `yarn lint:js --fix`.

### CODE REVIEW

**Comment by ef4** [comment]
Thanks.  A challenge here is that, as a deploy plugin, we don't have much in the way of test coverage. The `.travis.yml` file we have is vestigial, we're not actually running CI at all at the moment.

I can enable travis so that the linting will run, but we would need to clean up the other stuff that's in the travis.yml. There's no real "ember try" config, for example, because there's no ember app to test.

And for that reason, I'm not sure dependabot is a good idea for this repo. Without real functional test coverage, it could easily upgrade things into a broken state.

I just enabled travis for the repo, could you streamline the travis.yml in this PR and then we can see it turn green with just linting.

**Comment by dcyriller** [comment]
> And for that reason, I'm not sure dependabot is a good idea for this repo. Without real functional test coverage, it could easily upgrade things into a broken state.

Indeed, I'm removing it.

> I just enabled travis for the repo, could you streamline the travis.yml in this PR and then we can see it turn green with just linting.

Are you okay if I update it to run for node_js@8?

**Comment by dcyriller** [comment]
🍏

**Comment by ef4** [comment]
> Are you okay if I update it to run for node_js@8?

Definitely. No interest in node 0.12!

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,19 @@
 ---
 language: node_js
 node_js:
-  - "0.12"
+  - "8"
```
Comment (Turbo87) [inline .travis.yml]: @ef4 does this mean that older Node versions are no longer supported? if so this should have probably been released as a breaking change 🤔

Comment (dcyriller) [reply .travis.yml]: CI did not run at all. Also package.json states:
```
  "engines": {
    "node": ">= 4"
  },
```
 I guess it's only an improvement then.

Comment (Turbo87) [reply .travis.yml]: shouldn't we run CI with Node v4 then?

Comment (dcyriller) [reply .travis.yml]: I could add a Node@4 job if needed.

Comment (dcyriller) [reply .travis.yml]: 😂 you answered 9 seconds before me

**Pull request merged**